bigchaindb/backend/mongodb/connection.py

```diff
@@ -75,17 +112,35 @@ def _connect(self):
             # we should only return a connection if the replica set is
             # initialized. initialize_replica_set will check if the
             # replica set is initialized else it will initialize it.
-            initialize_replica_set(self.host, self.port, self.connection_timeout,
-                                   self.dbname, self.ssl, self.login, self.password)
-
-            # FYI: this might raise a `ServerSelectionTimeoutError`,
-            # that is a subclass of `ConnectionFailure`.
-            client = pymongo.MongoClient(self.host,
-                                         self.port,
-                                         replicaset=self.replicaset,
-                                         serverselectiontimeoutms=self.connection_timeout,
-                                         ssl=self.ssl)
-
+            initialize_replica_set(self.host,
+                                   self.port,
+                                   self.connection_timeout,
+                                   self.dbname,
+                                   self.ssl,
+                                   self.login,
+                                   self.password,
+                                   self.ca_cert,
+                                   self.certfile,
+                                   self.keyfile,
+                                   self.keyfile_passphrase,
+                                   self.crlfile)
+
+            # The presence of ca_cert, certfile, keyfile, crlfile implies the
+            # use of certificates for TLS connectivity.
+            # FYI: the connection process might raise a
+            # `ServerSelectionTimeoutError`, that is a subclass of
+            # `ConnectionFailure`.
+            if self.ca_cert is None or self.certfile is None or \
```

Comment:
That means the MongoDB instance will use the certificates from the OS to connect to the database.
The `self.ca_cert`, etc. kick in only for the self-signed certificates.
Source: [link](http://api.mongodb.com/python/current/examples/tls.html#specifying-a-ca-file)